### ISSUE TITLE
Test MueLu_ParameterListInterpreterTpetra_MPI_1 regularly randomly timing out in 'waterman' CUDA opt builds starting 2020-01-29

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#7090]
CC: @trilinos/muelu, @srajama1 (Trilinos Linear Solvers Product Lead)

<Checklist>
<???: Add label "client: ATDM">
<???: Add label "ATDM Sev: Blocker" (by default but could be other "ATDM Sev: XXX")>
<???: Add label "type: bug"?>
<???: Add label "impacting: tests"?>
<???: Add label for affected packages (e.g. "pkg: MueLu", "pkg: Tpetra", "pkg: Kokkos", etc.)>
<???: Add label "PA: ???Project Area???" (e.g. "PA: Linear Solvers", "PA: Data Services")>
<???: Add milestone "Initial cleanup of new ATDM ..." or "Keep promoted ATDM ...">
<???: Once GitHub Issue is created, add entries for tests to TrilinosATDMStatus/*.csv files>

## Next Action Status

Having this test run on its own GPU does not help (see PR #6840).  This test is disabled in all ATDM Trilinos 'waterman' CUDA builds in PR #7407 and has been merged to 'atdm-nightly' in commit 4804b08.  Next: Waiting for confirmation on CDash that this test is disabled in ATDM Trilinos CUDA builds starting testing day 2020-05-21 ...

## Description

As shown in [this query](https://testing-dev.sandia.gov/cdash/queryTests.php?project=Trilinos&begin=2019-10-01&end=2020-03-28&filtercount=5&showfilters=1&filtercombine=and&field1=groupname&compare1=62&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-waterman&field3=testname&compare3=65&value3=MueLu_ParameterListInterpreterTpetra_MPI_&field4=status&compare4=61&value4=failed&field5=details&compare5=63&value5=Timeout) the test:

* `MueLu_ParameterListInterpreterTpetra_MPI_1`

in the builds:

* `Trilinos-atdm-waterman-cuda-9.2-opt`
* `Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt`
* `Trilinos-atdm-waterman_cuda-9.2_shared_opt`

are fairly regularly randomly timing out starting on testing day <YYYY-MM-DD>.

The last 30 days of history are summarized in this table:

Site | Build Name | Test Name | Status | Details | Consec­utive Non-pass Days | Non-pass Last 30 Days | Pass Last 30 Days
-- | -- | -- | -- | -- | -- | -- | --
waterman | Trilinos-atdm-waterman-cuda-9.2-opt | MueLu_­ParameterListInterpreterTpetra_­MPI_­1 | Failed | Completed (Timeout) | 2 | 14 | 11
waterman | Trilinos-atdm-waterman_­cuda-9.2_­fpic_­static_­opt | MueLu_­ParameterListInterpreterTpetra_­MPI_­1 | Failed | Completed (Timeout) | 2 | 9 | 16  
waterman | Trilinos-atdm-waterman_­cuda-9.2_­shared_­opt | MueLu_­ParameterListInterpreterTpetra_­MPI_­1 | Failed | Completed (Timeout) | 22 | 24 | 1

## Current Status on CDash

* [Test results for this test in these builds last 30 days](https://testing-dev.sandia.gov/cdash/queryTests.php?project=Trilinos&begin=30%20days%20ago&end=now&filtercount=3&showfilters=1&filtercombine=and&field1=groupname&compare1=62&value1=Experimental&field2=block&field2count=3&field2field1=buildname&field2compare1=61&field2value1=Trilinos-atdm-waterman-cuda-9.2-opt&field2field2=buildname&field2compare2=61&field2value2=Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt&field2field3=buildname&field2compare3=61&field2value3=Trilinos-atdm-waterman_cuda-9.2_shared_opt&field3=testname&compare3=61&value3=MueLu_ParameterListInterpreterTpetra_MPI_1)

## Steps to Reproduce

One should be able to reproduce this failure on the machine 'waterman' as described in:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#waterman

The exact commands to reproduce this issue for one of these builds should be:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh \
    Trilinos-atdm-waterman-cuda-9.2-opt

$ cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_ALL_PACKAGES=ON \
 $TRILINOS_DIR

$ make NP=20

$ bsub -x -Is -n 20 ctest -j4
```

**Comment by bartlettroscoe** [comment]
NOTE: This is likely caused due to overloading the GPU similar to #6329, #6801, #6804, and #6805.

Therefore, I will work to resolve this along with these others by getting the tests to run better on the GPUs as per:

* https://gitlab.kitware.com/snl/project-1/-/issues/100

and RP #6802 

Assigning to myself and adding `ATDM Env Issue` label.

**Comment by bartlettroscoe** [comment]
This test is disabled in all ATDM Trilinos CUDA PR builds in PR #7407 and has been merged to 'atdm-nightly' in commit 4804b08.

Putting in review.
